pkg/dump/skip_defaults.go

```diff
@@ -0,0 +1,578 @@
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func removeDefaultsFromEntity(entity interface{}, entityType string, schemaFetcher *SchemaFetcher) error {
+	ptr := reflect.ValueOf(entity)
+	if ptr.Kind() != reflect.Ptr {
+		return fmt.Errorf("entity is not a pointer")
+	}
+
+	v := reflect.Indirect(ptr)
+
+	entityIdentifier, err := getEntityIdentifier(v, entityType)
+	if err != nil {
+		return fmt.Errorf("error getting entity identifier for schema fetching: %w", err)
+	}
+
+	schema, err := schemaFetcher.getSchema(entityType, entityIdentifier)
+	if err != nil {
+		return fmt.Errorf("error fetching schema for entity %s of type %s: %w", entityIdentifier, entityType, err)
+	}
+
+	defaultFields := make(map[string]interface{})
+	jsonb, err := json.Marshal(&schema)
+	if err != nil {
+		return err
+	}
+	gjsonSchema := gjson.ParseBytes((jsonb))
+
+	defaultFields = parseSchemaForDefaults(gjsonSchema, defaultFields)
+	if defaultFields == nil {
+		return fmt.Errorf("error parsing schema for defaults")
+	}
+
+	// no processing needed if no default fields found
+	if len(defaultFields) == 0 {
+		return nil
+	}
+
+	err = parseEntityWithDefaults(v, defaultFields)
+	if err != nil {
+		return fmt.Errorf("error parsing entity with defaults: %w", err)
+	}
+
+	return nil
+}
+
+func getEntityIdentifier(v reflect.Value, entityType string) (string, error) {
+	var zero reflect.Value
+	entityIdentifier := entityType
+
+	switch entityType {
+	case "plugins", "vaults":
+		name := v.FieldByName("Name")
+		if name == zero {
+			return "", fmt.Errorf("entity %s has no Name field for schema fetching", entityType)
+		}
+
+		entityIdentifier = *name.Interface().(*string)
+	case "partials":
+		partialType := v.FieldByName("Type")
+		if partialType == zero {
+			return "", fmt.Errorf("entity partial has no Type field for schema fetching")
+		}
+
+		entityIdentifier = *partialType.Interface().(*string)
+	}
+
+	return entityIdentifier, nil
+}
+
+func parseSchemaForDefaults(schema gjson.Result, defaultFields map[string]interface{}) map[string]interface{} {
+	schemaFields := schema.Get("fields")
+	defaultRecordValue := schema.Get("default")
+
+	schemaFields.ForEach(func(_, value gjson.Result) bool {
+		ms := value.Map()
+		fname := ""
+		for k := range ms {
+			fname = k
+			break
+		}
+
+		fieldSchema := value.Get(fname)
+		if fieldSchema.Get("fields").Exists() {
+			nestedMap := parseSchemaForDefaults(fieldSchema, make(map[string]interface{}))
+			if nestedMap == nil {
+				return false
+			}
+			defaultFields[fname] = nestedMap
+		}
+
+		if defaultRecordValue.Exists() && defaultRecordValue.Get(fname).Exists() {
+			value = defaultRecordValue.Get(fname)
+		} else {
+			value = value.Get(fname + ".default")
+		}
+
+		if value.Exists() {
+			defaultFields[fname] = value.Value()
+		}
+
+		return true
+	})
+
+	return defaultFields
+}
+
+func parseEntityWithDefaults(entity reflect.Value, defaultFields map[string]interface{}) error {
```

Comment:
nitpick - function name could indicate we are removing defaults wherever it makes sense?